library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsChunkSource.java

```diff
@@ -206,10 +206,9 @@ public void getNextChunk(HlsMediaChunk previous, long playbackPositionUs, HlsChu
     int oldVariantIndex = previous == null ? C.INDEX_UNSET
         : trackGroup.indexOf(previous.trackFormat);
     expectedPlaylistUrl = null;
-    // Use start time of the previous chunk rather than its end time because switching format will
-    // require downloading overlapping segments.
+    // Use end time instead because bufferedDurationUs will always be zero for 30s segment
```

Comment:
30s segments are not sensible, and it does not make sense to optimize for this case.

As an example of why 30s segments aren't sensible, see the comment that was previously here: switching variant in HLS can require downloading overlapping segments. A 30s segment duration implies you'd need to wastefully download 30s of overlapping media to switch variant; which is clearly very wasteful.